src/Pos/Genesis.hs

```diff
@@ -136,49 +159,36 @@ genesisUtxo bootStakeholdersMaybe ad
     balances = concatMap (uncurry zip) somethingComplicated
     utxoEntry (addr, coin) =
         ( TxIn (unsafeHash addr) 0
--- Note: it's quite bad because HD wallets become stakeholders and each
--- transaction will give them quite a lot of stake.
--- But CSL-1351 is planned for this sprint, so I suppose it's fine (it's
--- unlikely to cause problems anyway).
--- TODO [CSL-1351] Delete ↓.
         , TxOutAux (TxOut addr coin) (outDistr coin))
     -- Empty distribution for PubKey address means that the owner of
     -- this address will have the stake.
-    outDistr coin = maybe [] (flip genesisSplitBoot coin) bootStakeholdersMaybe
--- TODO [CSL-1351] Uncomment ↓.
---         , TxOutAux (TxOut addr coin) (genesisSplitBoot bootStakeholders coin))
---     bootStakeholdersCalculated = balancesToStakeholders balances
---     bootStakeholders =
---         fromMaybe bootStakeholdersCalculated bootStakeholdersMaybe
-
--- balancesToStakeholders :: [(Address, Coin)] -> HashMap StakeholderId Word16
--- balancesToStakeholders balances = foldr step mempty balances
```

Comment:
> No we don't. Why would we? What's the point?

1. Because that's how it works now and you shouldn't change this behavior unless you have a really good reason for it. Currently I don't see a good reason, I even implemented necessary logic.
2. It's counter-intuitive to use, for example, `ExponentialStakes` and see that all nodes have equal chance to become a slot leader. Really, the main idea of all these stake distributions was to affect stakes! For instance, `BitcoinDistribution` was exactly for this reason. `RichPoor` is also for stakes (we don't care who is rich and who is poor with respect to balances).
3. **Most important actually.** If the previous two points are not enough, I can give an example: people are using `RichPoorStakes` for benchmarks (in dev mode). There are a lot of poor nodes (50000) and some rich nodes. If all of them have equal stake, there will be no richmen, which is not what we want in benchmarks.